include/ht_singleton.h

```diff
@@ -14,27 +14,35 @@
 
 #pragma once
 
-#include <ht_platform.h>
-#include <ht_noncopy.h>
-
-namespace Hatchit {
-
-    namespace Core {
-
+//Header includes
+#include <ht_platform.h> //HT_API
+#include <ht_noncopy.h> //INonCopy
+
+//Inline includes
+#include <type_traits> //std::is_default_constructable
+
+namespace Hatchit
+{
+    namespace Core
+    {
+        /**
+        \class Singleton<T>
+        \ingroup HatchitCore
+        \brief Describes class that has only one instance avialable globally.
+
+        Describes class that has only one instance available globally.  Is
+        unable to be copied or instantiated elsewhere.
+        **/
```

Comment:
Can you just move this up just under the copyright notice? Thats where the other files have been putting this info